### DISCUSSION TITLE
Welcome to ROS-Mobile-Android Forum!

### DISCUSSION THREAD

**Discussion by nicostudt** [ROS-Mobile/ROS-Mobile-Android#76] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.

**Comment by tihobg** [comment]
Hi,

I want to control my ros robot using ros mobile app. I've install it from google play store. It's working now. I wanted to install it from source and developing the application using android studio. After installing the application on my pc, the build procedure has failed. This is my error:

Using insecure protocols with repositories, without explicit opt-in, is unsupported. Switch Maven repository 'maven2(http://repository.springsource.com/maven/bundles/release)' to redirect to a secure protocol (like HTTPS) or allow insecure protocols. See https://docs.gradle.org/7.0.2/dsl/org.gradle.api.artifacts.repositories.UrlArtifactRepository.html#org.gradle.api.artifacts.repositories.UrlArtifactRepository:allowInsecureProtocol for more details. 

Please, help!

**Reply by nicostudt** [reply]
Maybe you can get away with it: https://stackoverflow.com/questions/68585885/allow-insecure-protocols-android-gradle

**Reply by tihobg** [reply]
I tried all these suggestions but without success.

I tried making changes here in my build.gradle file:

allprojects {
    repositories {
        google()
        maven
                {
                    url 'https://repo1.maven.org/maven2'
                }

    }
}

I saw that the error contains a different http string: 
maven2(http://repository.springsource.com/maven/bundles/release)

Where I can find it? I think that I make changes at a wrong place. Please, help!

**Reply by nicostudt** [reply]
Since this project depends quite heavily on the implementation of the fairly outdated [rosjava](https://github.com/rosjava), you might have a better chance to resolve your problem by creating an issue at their repo. We are really unsure how to properly work around this kind of problems when it comes to newer Android versions without reimplementing rosjava by ourselves.

**Reply by tihobg** [reply]
What I can do in fact? Could you please suggest any another way to create an android application of my ros robot controlled through SSH protocol? Thank you in advance!

**Reply by nicostudt** [reply]
To my knowledge, rosjava is the only native way on Android to communicate with ROS. Otherwise, as a personal advice, you could also try to write your own web socket communication. There are some good tutorials out there for this.
Please keep me up to date!

**Comment by FaustineGitHub** [comment]
Excuse me,When I use the demo downloaded from https://github.com/ROS-Mobile/ROS-Mobile-Android and run it on Android studio, the error message is:
* What went wrong:
A problem occurred evaluating root project 'RosAndroid'.
> Could not get resource 'https://github.com/rosjava/rosjava_bootstrap/raw/kinetic/buildscript.gradle'.
    > Could not GET 'https://raw.githubusercontent.com/rosjava/rosjava_bootstrap/kinetic/buildscript.gradle'.
       > raw.githubusercontent.com

Excuse me, is there any solution?

**Reply by nicostudt** [reply]
Hey, as the rosjava repo is fairly outdated, the solution might be to downgrade your gradle version. Check issue #70 .This might hopefully help you to get it working!

**Reply by FaustineGitHub** [reply]
> Hey, as the rosjava repo is fairly outdated, the solution might be to downgrade your gradle version. Check issue #70 .This might hopefully help you to get it working!

Thank you very much for your reply. When I downgrade the gradle version to 3.0.1 as you suggested, the same problem still occurs. Is there anything else to pay attention to when importing the project into Android studio?
A problem occurred evaluating root project 'RosAndroid'.
> Could not read script 'https://github.com/rosjava/rosjava_bootstrap/raw/kinetic/buildscript.gradle'.
    > Read timed out

**Reply by nicostudt** [reply]
This seems to be a common problem with the rosjava package 😑 I also cannot reach the link as something changed there. I'll further inspect that!

Also I found a similar issue here: https://github.com/rosjava/android_core/issues/282

Do you have a working ROS distribution installed on your OS?

**Reply by nicostudt** [reply]
Check if you can build the project via VPN if you are located in China or other asian contries that might block some webpages.

**Reply by FaustineGitHub** [reply]
> distribution

I don't have ROS distribution installed on my computer. This is the first time I come into contact with ROS Android-related things. I don't understand many things. There is still a lot to learn. If it is convenient, please give me some pointers!

**Reply by nicostudt** [reply]
Thats okay. No problem! 
What do you want to do with the app? 
There are some good instructions on the official ROS Website. I personally recommend using the Melodic distribution, if your are unsure, which one to use.

**Reply by FaustineGitHub** [reply]
> Thats okay. No problem! What do you want to do with the app? There are some good instructions on the official ROS Website. I personally recommend using the Melodic distribution, if your are unsure, which one to use.

I see that the description of kinetic on the official ROS website is more detailed, and your ROS-Mobile-Android also calls https://github.com/rosjava/android_core/raw/kinetic/buildscript.gradle
The first problem I want to solve now is to successfully run the ROS-Mobile-Android project on Android studio, but the resource file keeps failing to load, I can't find the reason here, thank you for your advice, I will continue to work hard of

**Comment by otorunoglu** [comment]
Hi,
Thanks for this application. This is a very powerful application. 
We want to use with **mobile internet**, because we use our ros robot where there is no wifi connection.
If we connect with WIFI, everything is fine. 
We connected the robot to the mobile phone with the USB connection. We get the Internet from the mobile network of the mobile phone.Mobile network is private 5G mmWave internet. 
However, we could not connect to robot with the application over **mobile network**.
Can we connect over the **mobile network**? How can we do that?

**Reply by nicostudt** [reply]
Hey, thank you for your feedback!
Try the [tutorial](https://nrottmann.github.io/tutorials/raspberrypi/#Web) for using ROS with mobile internet by one of our authors (Rottmann).
Otherwise you may be more successful if you ask in the official ROS forum.

Best wishes

**Reply by otorunoglu** [reply]
Thanks for the answer.

**Comment by otorunoglu** [comment]
Hi,
I want to ask about using to joystick for different tasks at same time?
We use omni robot. 
We want to use two virtual joysticks at the same time. 
The first joystick will move in x and y axis, the second joystick will make circular movement.
However, when we set the parameters of the two joysticks, they both provide the same movement even though the parameters are different.
How can I do two different movements on two separate joysticks?
Thanks.

**Reply by nicostudt** [reply]
Hey,
you can achieve that by using two joystick widgets, which are publishing to two different topics and then merge them together again on the ROS side. There was an issue, where this kind of behavior was discussed (#89). 
If you try to publish to the same topic with multiple widgets simultaneously, the message on the topic will be overwritten.

I hope I was able to help! 
Best regards

**Reply by otorunoglu** [reply]
Thanks for the answer. I'll try changing topic names.
